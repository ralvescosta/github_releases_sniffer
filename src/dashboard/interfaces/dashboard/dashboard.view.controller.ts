import {useContext, useEffect, useState} from 'react';

import {SniffedRepositoriesContext} from '../../../core/context/sniffed.repositories.context';
import {IGetSnifferReposAndAccountUsecase} from '../../bussiness/usecases/iget.sniffer.repos.and.account.usecase';

export class DashboardViewController {
  public context = useContext(SniffedRepositoriesContext);
  public userAccount: any;
  public setUserAccount: React.Dispatch<React.SetStateAction<any>>;

  constructor(private readonly _getSnifferReposAndAccountUsecase: IGetSnifferReposAndAccountUsecase) {
    [this.userAccount, this.setUserAccount] = useState({});

    useEffect(() => {
      this.getSniffedReposAndAccountIntoCacheAndSetGlobalContext();
    }, []);
  }

  private async getSniffedReposAndAccountIntoCacheAndSetGlobalContext() {
    const result = await this._getSnifferReposAndAccountUsecase.get();
    this.context.setSniffedRepositories(result.sniffer);
    this.setUserAccount(result.user);
  }
}
