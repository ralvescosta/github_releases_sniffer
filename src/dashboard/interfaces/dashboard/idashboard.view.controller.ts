import React from 'react';

export interface IDashboardViewController {
  context: any;
  userAccount: any;
  setUserAccount: React.Dispatch<React.SetStateAction<any>>;
}
