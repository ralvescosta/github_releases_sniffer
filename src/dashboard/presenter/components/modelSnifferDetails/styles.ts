import {StyleSheet} from 'react-native';
import {widthToDP} from '../../../../core/themes/size';

export const styles = StyleSheet.create({
  modalContainer: {
    flex: 1,
  },
  closeModal: {
    padding: widthToDP('3%'),
  },
  containerContent: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around',
  },

  contentRepository: {
    width: '100%',
    alignItems: 'center',
  },
  repositoryOwnerAvatar: {
    width: 85,
    height: 85,
  },
  repositoryOwnerFakeAvatar: {
    width: 85,
    height: 85,
    backgroundColor: '#ccc',
  },
  repositoryText: {
    fontSize: widthToDP('8%'),
    fontWeight: 'bold',
    marginTop: widthToDP('2%'),
  },

  contentRelease: {
    width: '100%',
    alignItems: 'center',
  },
  releaseTitle: {
    fontSize: widthToDP('4.8%'),
  },
  releaseText: {
    fontSize: widthToDP('4.8%'),
    fontWeight: 'bold',
  },

  contentRemove: {
    width: '100%',
    height: 50,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 5,
    backgroundColor: '#fff',
  },
});
