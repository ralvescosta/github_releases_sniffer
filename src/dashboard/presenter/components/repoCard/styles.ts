import {StyleSheet} from 'react-native';
// import {primary, secondary} from '../../../../core/themes/colors';
import {widthToDP} from '../../../../core/themes/size';

export const styles = StyleSheet.create({
  repoCard: {
    width: '100%',
    flexDirection: 'row',
    padding: widthToDP('2%'),
    backgroundColor: '#fff',
    elevation: 5,
    borderRadius: 6,
    marginTop: widthToDP('2%'),
    // borderWidth: 2,
    // borderColor: '#000',
  },

  cardLeft: {
    maxWidth: '60%',
    width: '100%',
    // borderWidth: 2,
    // borderColor: '#000',
  },

  leftHeader: {},
  headerName: {
    fontWeight: 'bold',
  },
  headerDescription: {
    color: '#777',
    marginTop: widthToDP('0.5%'),
  },

  leftContent: {
    flexDirection: 'row',
    marginTop: widthToDP('2%'),
  },
  contentDetails: {
    flexDirection: 'row',
    alignItems: 'center',
    marginRight: 7,
    color: '#777',
  },
  detailsText: {
    fontSize: widthToDP('3.4%'),
    marginLeft: 3,
    maxWidth: 40,
  },

  cardRight: {
    maxWidth: '40%',
    width: '100%',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    // borderWidth: 2,
    // borderColor: '#000',
  },
  rightRelease: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  releaseTitle: {
    fontWeight: 'bold',
  },
  releaseTag: {
    color: '#777',
  },
});
