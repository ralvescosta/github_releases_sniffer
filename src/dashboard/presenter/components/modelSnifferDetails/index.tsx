import React from 'react';
import {Modal, View, TouchableOpacity, Text, Image} from 'react-native';

import {styles} from './styles';

import {IModalSnifferDetailsViewController} from '../../../interfaces/modalSnifferDetails/imodal.sniffer.details.view.controller';
import {SniffedGithubRepositoryEntity} from '../../../bussiness/entities/sniffed.github.repository.entity';

type Props = {
  repository: SniffedGithubRepositoryEntity;
  viewController: IModalSnifferDetailsViewController;
};

export const ModelSnifferDetails = ({repository, viewController}: Props) => {
  return (
    <Modal animationType="slide" visible={viewController.modalContext.toggleModal}>
      <View style={styles.modalContainer}>
        <TouchableOpacity style={styles.closeModal} onPress={() => viewController.closeModal()}>
          <Text>X</Text>
        </TouchableOpacity>
        <View style={styles.containerContent}>
          <View style={styles.contentRepository}>
            <Image style={styles.repositoryOwnerAvatar} source={{uri: repository.ownerAvatarUrl}} resizeMode="cover" />
            <Text style={styles.repositoryText}>{repository.fullName}</Text>
          </View>

          <View style={styles.contentRelease}>
            <Text style={styles.releaseTitle}> Latest release: </Text>
            <Text style={styles.releaseText}>{repository.lastRelease}</Text>
          </View>

          <TouchableOpacity onPress={() => viewController.removeRepository(repository.id)} style={styles.contentRemove}>
            <Text>Remove Repository on Sniffer</Text>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  );
};
