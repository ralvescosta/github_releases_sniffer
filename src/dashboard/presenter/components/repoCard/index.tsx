import React from 'react';
import {View, Text, Alert} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

import {styles} from './styles';

type Props = {
  repository: any;
};

export const RepoCard = ({repository}: Props) => {
  function pressedCard() {
    Alert.alert('Github Sniffer', 'Do you really want removing this repository sniffed?', [
      {
        text: 'cancel',
        style: 'cancel',
      },
      {
        text: 'OK',
        onPress: async () => {},
      },
    ]);
  }

  return (
    <TouchableOpacity style={styles.repoCard} onPress={pressedCard}>
      <View style={styles.cardLeft}>
        <View style={styles.leftHeader}>
          <Text numberOfLines={2} style={styles.headerName}>
            {repository.fullName}
          </Text>
          <Text numberOfLines={3} ellipsizeMode="tail" style={styles.headerDescription}>
            {repository.description}
          </Text>
        </View>

        <View style={styles.leftContent}>
          <View style={styles.contentDetails}>
            <FontAwesome name="star" color="#777" size={15} />
            <Text numberOfLines={1} style={styles.detailsText}>
              {repository.stargazersCount}
            </Text>
          </View>

          <View style={styles.contentDetails}>
            <FontAwesome name="code-fork" color="#777" size={15} />
            <Text numberOfLines={1} style={styles.detailsText}>
              {repository.forks}
            </Text>
          </View>

          <View style={styles.contentDetails}>
            <FontAwesome name="bug" color="#777" size={15} />
            <Text numberOfLines={1} style={styles.detailsText}>
              {repository.openIssues}
            </Text>
          </View>
        </View>
      </View>

      <View style={styles.cardRight}>
        <View style={styles.rightRelease}>
          <Text style={styles.releaseTitle}>Last Release: </Text>
          <Text style={styles.releaseTag}>{repository.lastRelease}</Text>
        </View>
        <FontAwesome name="bell" color="#d4d4d4" size={30} />
      </View>
    </TouchableOpacity>
  );
};
