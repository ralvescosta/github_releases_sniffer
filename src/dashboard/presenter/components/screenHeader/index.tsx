import React from 'react';
import {View, Text, Image, TouchableOpacity} from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

import {styles} from './styles';

export const Header: React.FC = () => {
  return (
    <View style={styles.header}>
      <View style={styles.resumeColumn}>
        <View style={styles.resumeContent}>
          <FontAwesome name="code-fork" color="#fff" size={15} />
          <Text style={styles.resumeText}>Configured Repos: 10</Text>
        </View>
        <View style={styles.resumeContent}>
          <FontAwesome name="bell" color="#fff" size={15} />
          <Text style={styles.resumeText}>Alerts last Week: 10</Text>
        </View>
      </View>

      <TouchableOpacity style={styles.accountColumn} activeOpacity={0.4}>
        <Image style={styles.accountAvatar} source={{uri: 'https://avatars0.githubusercontent.com/u/45952161?v=4'}} />
        <Text style={styles.accountUser}>ralvescosta</Text>
      </TouchableOpacity>
    </View>
  );
};
