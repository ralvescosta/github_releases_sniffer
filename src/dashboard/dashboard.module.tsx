import React from 'react';
import {Dashboard} from './presenter/dashboard.screen';

import {DashboardViewController} from './interfaces/dashboard/dashboard.view.controller';

import {GetSnifferReposAndAccountUsecase} from './application/usecases/get.sniffer.repos.and.account.usecase';
import {LocalDatabaseRepository} from './infrastructure/repositories/local.database.repository';

import {RepoCard} from './presenter/components/repoCard';
import {RepoCardViewController} from './interfaces/repoCard/repo.card.view.controller';

import {ModelSnifferDetails} from './presenter/components/modelSnifferDetails';
import {ModalSnifferDetailsViewController} from './interfaces/modalSnifferDetails/modal.sniffer.details.view.controller';

const WrapperModelSnifferDetails = ({repository, modalControl, setModalControl}: any) => {
  const modalSnifferDetailsViewController = new ModalSnifferDetailsViewController(modalControl, setModalControl);
  return <ModelSnifferDetails repository={repository} viewController={modalSnifferDetailsViewController} />;
};

const WrapperRepoCard = ({repository}: any) => {
  const repoCardViewController = new RepoCardViewController();
  return <RepoCard repository={repository} ModelSnifferDetails={WrapperModelSnifferDetails} viewController={repoCardViewController} />;
};

export const DashboardModule = () => {
  const localDatabaseRepository = new LocalDatabaseRepository();
  const getSnifferReposAndAccountUsecase = new GetSnifferReposAndAccountUsecase(localDatabaseRepository);
  const dashboardViewController = new DashboardViewController(getSnifferReposAndAccountUsecase);

  return <Dashboard viewController={dashboardViewController} RepoCard={WrapperRepoCard} />;
};
