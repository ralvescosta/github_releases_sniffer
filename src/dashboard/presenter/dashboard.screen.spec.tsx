import React from 'react';
import {render} from '@testing-library/react-native';

import {Dashboard} from './dashboard.screen';
import {SniffedGithubRepositoryEntity} from '../bussiness/entities/sniffed.github.repository.entity';
import {RepoCardViewControllerSpy} from './__test__/repo.card.view.controller.spy';
import {ModalSnifferDetailsViewControllerSpy} from './__test__/modal.sniffer.details.view.controller.spy';
import {DashboardViewControllerSpy} from './__test__/dashboard.view.controller.spy';

import {ModelSnifferDetails} from './components/modelSnifferDetails';
import {RepoCard} from './components/repoCard';

function makeSut() {
  const Wrapper = () => {
    const repository = new SniffedGithubRepositoryEntity(1, '', '', 10, 10, 10, 1, '', '', '');
    const cardControllerSpy = new RepoCardViewControllerSpy();
    const modalControllerSpy = new ModalSnifferDetailsViewControllerSpy();
    const dashboardControllerSpy = new DashboardViewControllerSpy();

    const WrapperModal = () => {
      return <ModelSnifferDetails repository={repository} viewController={modalControllerSpy} />;
    };
    const WrapperRepoCard = () => (
      <RepoCard ModelSnifferDetails={WrapperModal} repository={repository} viewController={cardControllerSpy} />
    );
    return <Dashboard RepoCard={WrapperRepoCard} viewController={dashboardControllerSpy} />;
  };

  return render(<Wrapper />);
}

describe('Dashboard Screen', () => {
  it('Should render correctly', () => {
    const sut = makeSut();

    expect(sut.toJSON()).toMatchSnapshot();
  });
});
