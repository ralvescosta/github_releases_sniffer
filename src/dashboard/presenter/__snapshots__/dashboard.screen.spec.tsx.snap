// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Screen Should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f4f4f4",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#24292e",
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 22.5,
        "paddingVertical": 11.5,
        "width": "100%",
      }
    }
  >
    <View
      style={Object {}}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "color": "#e1e4e8",
              "marginLeft": 26.5,
            }
          }
        >
          Configured Repos: 
          0
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "color": "#e1e4e8",
              "marginLeft": 26.5,
            }
          }
        >
          Alerts last Week: 
          0
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 26.5,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 26.5,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 15,
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#e1e4e8",
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "paddingBottom": 35,
      }
    }
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;
